version: '3.7'
services:
  grafana:
    image: grafana/grafana-oss:latest
    container_name: zbxgrafana
    restart: always
    ports:
      - "38000:3000"
    environment:
      - GF_SERVER_ROOT_URL=http://grafana:3000
      - GF_LOG_LEVEL=debug
      - GF_FEATURE_TOGGLES_ENABLE=publicDashboards
      - GF_PLUGIN_GRAFANA_IMAGE_RENDERER_RENDERING_IGNORE_HTTPS_ERRORS=true
      - GF_DATABASE_SSL_MODE=disable
      - GF_DATABASE_TYPE=postgres
      - GF_DATABASE_HOST=grafanadb:5432
      - GF_DATABASE_USER=admin
      - GF_DATABASE_NAME=grafana    
    volumes:
      - 'grafana-data:/var/lib/grafana'
      - ./config/grafana.ini:/etc/grafana/grafana.ini
      - ./plugins:/var/lib/grafana/plugins      
    healthcheck:
      test: ["CMD", "sh", "-c", "ps aux | grep grafana || exit 1"]
      interval: 20s
      timeout: 10s
      retries: 2
      start_period: 5s
    networks:
      - zbx
  grafanadb:
    image: postgres:latest
    container_name: zbxgrafanadb
    restart: always
    environment:
      - POSTGRES_USER_FILE=/run/secrets/grafana_db_user
      - POSTGRES_PASSWORD_FILE=/run/secrets/grafana_db_pass
      - POSTGRES_DB=grafana
    secrets:
      - grafana_db_user
      - grafana_db_pass
    volumes:
      - 'postgres-data:/var/lib/postgresql/data'
    healthcheck:
      test: ["CMD", "pg_isready"]
      interval: 20s
      timeout: 10s
      retries: 2
      start_period: 5s
    networks:
      - zbx
volumes:
  postgres-data: {}
  grafana-data: {}
networks:
  zbx:
    external: true
secrets:
  grafana_db_user:
    file: ./config/.grafana_db_user
  grafana_db_pass:
    file: ./config/.grafana_db_pass
